lib/sycamore/sycamore/transforms/text_extraction/text_extractor.py

```diff
@@ -0,0 +1,13 @@
+from typing import List, Union
+from abc import abstractmethod
+from sycamore.data import Element
+from io import IOBase
+
+
+class TextExtractor:
+    @abstractmethod
+    def extract(self, filename: Union[str, IOBase], hash_key: str, use_cache=False, **kwargs) -> List[List[Element]]:
+        pass
+
+    def __name__(self):
```

Comment:
What's __name__ for? Doesn't look like you are using it or overriding it anywhere.

Reply:
Good point, I overrided it so that the objects can be identified from the superclass